pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -272,7 +274,8 @@ func (c *NPLController) isNPLEnabledForServiceOfPod(obj interface{}) bool {
 		svc, isSvc := service.(*corev1.Service)
 		// Selecting Services NOT of type NodePort, with Service selector matching Pod labels.
 		if isSvc && svc.Spec.Type != corev1.ServiceTypeNodePort {
-			if matchSvcSelectorPodLabels(svc.Spec.Selector, pod.GetLabels()) {
+			if matchSvcSelectorPodLabels(svc.Spec.Selector, pod.GetLabels()) &&
+				pod.GetNamespace() == svc.GetNamespace() {
```

Comment:
not really specific to this PR, but is there a difference between using the getters (e.g. `pod.GetNamespace()`) or accessing the fields directly (e.g. `pod.Namespace`)? @tnqn do you know?

Reply:
I think no real difference, but I woud just use pod.Namespace.

Reply:
I agree, it would be good to make it uniform and stop using the getters in the NPL controller. We don't use them anywhere else.

Reply:
The GetXXX methods are normally used when a function needs to be generic to handle different kinds of objects. The object is then converted to or passed in as metav1.Object